06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js

```diff
@@ -0,0 +1,174 @@
+const empty = require('../files/empty.json');
+
+const invertedIndex = new Index();
+
+const work = [{
+  title: 'Checkpoint one',
+  text: 'Testing getIndex function',
+},
+
+{
+  title: 'Checkpoint Two',
+  text: 'Coming soon',
+},
+];
+
+describe('Define methods for Index class', () => {
+  invertedIndex.createIndex('work.json', workBook);
+  it('Should ensure that invertedIndex methods are defined', () => {
+    expect(invertedIndex.getIndex).toBeDefined();
+  });
+  it('Should ensure that invertedIndex methods are defined', () => {
+    expect(invertedIndex.createIndex).toBeDefined();
+  });
+  it('Should ensure that invertedIndex methods are defined', () => {
+    expect(invertedIndex.tokenize).toBeDefined();
+  });
+  it('Should ensure that invertedIndex methods are defined', () => {
+    expect(Index.validateFile).toBeDefined();
+  });
+  it('Should ensure that invertedIndex methods are defined', () => {
+    expect(invertedIndex.searchIndex).toBeDefined();
+  });
+});
+
+describe('Should return an object that is an index of the file', () => {
+  invertedIndex.createIndex('work.json', work);
+  it('should verify that index has been created', () => {
+    expect(Object.keys(invertedIndex.getIndex('work.json', work)).length)
+      .toBeGreaterThan(0);
+  });
+
+  it('Should return an object that is an accurate index of the work.json file',
+    () => {
+      expect(invertedIndex.getIndex('work.json', work)).toEqual({
+        checkpoint: [0, 1],
+        one: [0],
+        testing: [0],
+        getindex: [0],
+        function: [0],
+        two: [1],
+        coming: [1],
+        soon: [1],
+      });
+    });
+});
+describe('Should create an object that is an accurate index of file', () => {
+  invertedIndex.createIndex('test.json', invalidBook);
+  it('should verify that index has been created', () => {
+    expect(Object.keys(invertedIndex.getIndex('test.json', invalidBook)).length)
+      .toBeGreaterThan(0);
+  });
+
+  it('Should create an object that is an accurate index of test.json file',
+    () => {
+      expect(invertedIndex.getIndex('test.json', invalidBook)).toEqual({
+        honey: [0],
+        i: [0],
+        shrunked: [0],
+        the: [0],
+        kids: [0],
+        an: [0],
+        american: [0],
+        movie: [0, 1],
+        produced: [0],
+        birds: [0],
+        ant: [1],
+        bully: [1],
+        animated: [1],
+        epic: [1],
+        of: [1],
+        ants: [1],
+        bullying: [1],
+        humans: [1],
+      });
+    });
+});
+
+describe('Search Index', () => {
+  it('Should return correct index of the search term in work.json', () => {
+    expect(invertedIndex.searchIndex('Testing, soon', 'work.json')).toEqual({
+      testing: [0],
+      soon: [1],
+    });
+  });
+  it('should return true if search term is a string', () => {
+    const words = 'Alice in Wonderland';
+    expect(Object.keys(invertedIndex.searchIndex('words'))).toBeTruthy();
```

Comment:
Are you meant to pass a string directly here or **_words_** which you defined on line 99 of this file?